zerver/tests/test_queue_worker.py

```diff
@@ -536,12 +531,12 @@ def consume(self, data: Mapping[str, Any]) -> None:
         with simulated_queue_client(lambda: fake_client):
             worker = UnreliableWorker()
             worker.setup()
-            with patch('logging.exception') as logging_exception_mock:
+            with self.assertLogs(level='ERROR') as m:
                 worker.start()
-                logging_exception_mock.assert_called_once_with(
-                    "Problem handling data on queue %s", "unreliable_worker",
-                    stack_info=True,
-                )
+                self.assertEqual(m.records[0].message, 'Problem handling data on queue unreliable_worker')
+                # splitting m.output to remove stack information of test case and to get,
+                # stack frame of actual exception causing function.
+                self.assertEqual(m.output[0].split('Exception: Worker task not performing as expected!\n')[1], m.records[0].stack_info)
```

Comment:
I'm unable to understand this. What is it that we are expecting `m.records[0].stack_info` to have?

Reply:
Here we expect the `m.records[0].stack_info` to have stack traceback of the error thrown. As m.output[0] also contains information about the testcase so I had to split that part as `Traceback` will contain the trace from where error was thrown and not test case information.

Reply:
Ohh,  should we try using `assertRegexMatches` or `assertIn` as `'Exception: Worker task not performing as expected!\n'` seems to be unrelated to test and is confusing?